Update example in “Inheritance and the prototype chain”

muditbhadouria opened the pull request:
<!-- Please provide the following information to help us review this PR: -->

> What was wrong/why is this fix needed? (quick summary only)
The existing example assigns a new object to the prototype property of Graph constructor function.
Graph.prototype = {
  addVertex: function(v) {
    this.vertices.push(v);
  }
};
This is not a good approach. Instead, the addVertex function should be added as a new property in the existing prototype object.
Graph.prototype.addVertex = function(v) {
  this.vertices.push(v);
}


> Issue number (if there is an associated issue)
NA

> Anything else that could help us review it
NA

muditbhadouria committed:
Updating the example for creating an object with a constructor

The existing example assigns a new object to the prototype property of Graph constructor function.
Graph.prototype = {
  addVertex: function(v) {
    this.vertices.push(v);
  }
};
This is not a good approach. Instead, the addVertex function should be added as a new property in the existing prototype object.
Graph.prototype.addVertex = function(v) {
  this.vertices.push(v);
}

s-mrb reviewed (approved):
seems good.

@muditbhadouria it would be great if you also make edit on line 250,

> "constructor" in JavaScript is "just"

and change it to

> <strong>constructor</strong> in JavaScript is <strong>just</strong>

via `<strong></strong>` tag

muditbhadouria committed:
Substituting the quotes with bold characters

As per review to the previous commit, replaced the double quotes with bold characters

muditbhadouria commented:
Found multiple instances which require the described change (e.g. line 391). Should I go ahead with them in the same PR? Few instances (e.g. line 395) where the constructor function is not called while instantiating it using the new keyword. Can I change them as well

sideshowbarker commented:
> Found multiple instances which require the described change (e.g. line 391). Should I go ahead with them in the same PR? Few instances (e.g. line 395) where the constructor function is not called while instantiating it using the new keyword. Can I change them as well?

Yes, please go ahead with adding those changes on this same PR branch

sideshowbarker commented on files/en-us/web/javascript/inheritance_and_the_prototype_chain/index.html:
```diff
@@ -247,18 +247,16 @@ <h3 id="Objects_created_with_syntax_constructs">Objects created with syntax cons
 <h3 id="With_a_constructor">With a constructor</h3>
 
-<p>A "constructor" in JavaScript is "just" a function that happens to be called with the <a href="/en-US/docs/Web/JavaScript/Reference/Operators/new">new operator</a>.</p>
+<p>A <strong>constructor</strong> in JavaScript is <strong>just</strong> a function that happens to be called with the <a href="/en-US/docs/Web/JavaScript/Reference/Operators/new">new operator</a>.</p>
```
```suggestion
<p>A constructor in JavaScript is just a function that happens to be called with the <a href="/en-US/docs/Web/JavaScript/Reference/Operators/new">new operator</a>.</p>
```

sideshowbarker committed:
Update files/en-us/web/javascript/inheritance_and_the_prototype_chain/index.html

sideshowbarker commented:
> > Found multiple instances which require the described change (e.g. line 391). Should I go ahead with them in the same PR? Few instances (e.g. line 395) where the constructor function is not called while instantiating it using the new keyword. Can I change them as well?
> 
> Yes, please go ahead with adding those changes on this same PR branch

Actually, nevermind — that change is unnecessary. There’s no need for any special styling of those words in this case.

sideshowbarker commented:
> Found multiple instances which require the described change (e.g. line 391). Should I go ahead with them in the same PR? Few instances (e.g. line 395) where the constructor function is not called while instantiating it using the new keyword. Can I change them as well?

Yes, but let’s go ahead and merge this PR first, and then you can open a separate new PR for the further changes.

sideshowbarker commented:
Mudit, thanks much, and congrats on landing your first docs change here — welcome aboard 🎉

muditbhadouria commented:
Thank you 😀
